Time history for ABL forcing

locascio-m opened the pull request:
Added input table for ABL forcing time, wind speed, and wind direction

tonyinme commented:
hey @locascio-m , the PR is looking good. Can you please add the documentation as well?

locascio-m commented:
My mistake, just added it

locascio-m commented:
I adjusted the code in ABLForcing.cpp based on the warning from Lint-cppcheck. The other warnings there and the error in the Docker check are not related to code that I've modified though.

gdeskos commented on amr-wind/equation_systems/icns/source_terms/ABLForcing.cpp:
```diff
@@ -19,7 +20,26 @@ ABLForcing::ABLForcing(const CFDSim& sim) : m_time(sim.time())
+
+    pp_abl.query("velocity_timetable", m_vel_timetable);
+    if (!m_vel_timetable.empty()) {
+        std::ifstream ifh(m_vel_timetable, std::ios::in);
```
Make sure you include something to do the end of line. See here https://github.com/Exawind/amr-wind/blob/3d2e837f51a8c3a403c359d40264b21bf5c88850/amr-wind/utilities/tagging/CartBoxRefinement.cpp#L38

gdeskos commented on amr-wind/wind_energy/ABLFieldInit.cpp:
```diff
@@ -63,9 +77,20 @@ void ABLFieldInit::operator()(
+    amrex::Real vmean;
+    amrex::Real wmean;
+    if (!m_vel_timetable.empty()) {
+        umean = m_vel_speed * std::cos(m_vel_dir);
```
turn these back to const amrex::Real

locascio-m commented:
Added a new commit to address @gdeskos comments

tonyinme commented on docs/sphinx/user/inputs_Momentum_Sources.rst:
```diff
@@ -67,3 +67,10 @@ Section: Momentum Sources
+
+   **type:** String, optional
+   
+   Input file name for table that lists time, wind speed, and wind direction
```
Can you specify the format of the table and the units of each variable?

tonyinme reviewed (commented):
This looks good. Please address the suggestion in the documentation.
